analyzers/tests/SonarAnalyzer.Test/Rules/MethodsShouldNotHaveIdenticalImplementationsTest.cs

```diff
@@ -22,36 +22,151 @@
+        builderCS.AddSnippet($$"""
+            using System;
+            public static class TypeConstraints
+            {
+                public static bool Compare1<T>(T? value1, T value2) {{constraint1}}
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+
+                public static bool Compare2<T>(T? value1, T value2) {{constraint2}}
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+            }
+            """).WithOptions(ParseOptionsHelper.FromCSharp9).VerifyNoIssues();
+
+    [DataTestMethod]
+    [DataRow("", "")]
+    [DataRow("where TKey: struct", "where TKey: struct")]
+    [DataRow("where TKey: struct where TValue: class", "where TKey: struct where TValue: class")]
+    [DataRow("where TValue: class where TKey: struct", "where TKey: struct where TValue: class")]
+    [DataRow("where TKey: class", "where TKey: class")]
+    [DataRow("where TKey: unmanaged", "where TKey: unmanaged")]
+    [DataRow("where TKey: new()", "where TKey: new()")]
+    [DataRow("where TKey: IEquatable<TKey>, IComparable", "where TKey: System.IComparable, IEquatable<TKey>")]
+    [DataRow("where TKey: IEquatable<TKey>, IComparable where TValue: IComparable", " where TValue: System.IComparable where TKey: System.IComparable, IEquatable<TKey>")]
+    public void MethodsShouldNotHaveIdenticalImplementations_MethodTypeParameters_Dictionary_NonCompliant(string constraint1, string constraint2) =>
+        builderCS.AddSnippet($$"""
+            using System;
+            using System.Collections.Generic;
+            public static class TypeConstraints
+            {
+                public static bool Test1<TKey, TValue>(IDictionary<TKey, TValue> dict) {{constraint1}} // Secondary
+                {
+                    Console.WriteLine(dict);
+                    Console.WriteLine(dict);
+                    return true;
+                }
+
+                public static bool Test2<TValue, TKey>(IDictionary<TKey, TValue> dict) {{constraint2}} // Noncompliant
+                {
+                    Console.WriteLine(dict);
+                    Console.WriteLine(dict);
+                    return true;
+                }
+            }
+            """).WithOptions(ParseOptionsHelper.FromCSharp9).Verify();
+
+    [DataTestMethod]
+    [DataRow("")]
+    [DataRow("where TKey: struct")]
+    [DataRow("where TKey: struct where TValue: class")]
+    [DataRow("where TValue: class where TKey: struct")]
+    [DataRow("where TKey: class")]
+    [DataRow("where TKey: unmanaged")]
+    [DataRow("where TKey: new()")]
+    [DataRow("where TKey: IEquatable<TKey>, IComparable")]
+    [DataRow("where TKey: IEquatable<TKey>, IComparable where TValue: IComparable")]
```

Comment:
All test cases with only one type parameters seem to duplicate `MethodsShouldNotHaveIdenticalImplementations_MethodTypeParameters_NonCompliant`.

Reply:
This test is about type parameter originating from the class instead of the method